model/task/task.go

```diff
@@ -328,45 +328,41 @@ type StepbackInfo struct {
 	// PreviousStepbackTaskId stores the last stepback iteration id.
 	PreviousStepbackTaskId string `bson:"previous_stepback_task_id,omitempty" json:"previous_stepback_task_id"`
 	// GeneratedStepbackInfo stores information on a generator for it's generated tasks.
-	GeneratedStepbackInfo []GeneratedStepbackInfo `bson:"generated_stepback_info,omitempty" json:"generated_stepback_info,omitempty"`
-}
+	GeneratedStepbackInfo []StepbackInfo `bson:"generated_stepback_info,omitempty" json:"generated_stepback_info,omitempty"`
 
-type GeneratedStepbackInfo struct {
+	// Generator fields only (responsible for propogating stepback in its generated tasks).
 	// DisplayName is the display name of the generated task.
 	DisplayName string `bson:"display_name,omitempty" json:"display_name,omitempty"`
 	// BuildVariant is the build variant of the generated task.
 	BuildVariant string `bson:"build_variant,omitempty" json:"build_variant,omitempty"`
-
-	// See StepbackInfo for more information on these fields. These fields specifically
-	// store the task ids for the generator tasks.
-	LastFailingStepbackTaskId string `bson:"last_failing_stepback_task_id,omitempty" json:"last_failing_stepback_task_id"`
-	LastPassingStepbackTaskId string `bson:"last_passing_stepback_task_id,omitempty" json:"last_passing_stepback_task_id"`
-	NextStepbackTaskId        string `bson:"next_stepback_task_id,omitempty" json:"next_stepback_task_id"`
-	PreviousStepbackTaskId    string `bson:"previous_stepback_task_id,omitempty" json:"previous_stepback_task_id"`
 }
 
+// IsZero returns true if the StepbackInfo is empty or nil.
+// It does not include GeneratedStepbackInfo in the check because
+// those do not cause a generator to stepback.
 func (s *StepbackInfo) IsZero() bool {
 	if s == nil {
 		return true
 	}
 	if s.LastFailingStepbackTaskId != "" && s.LastPassingStepbackTaskId != "" {
 		return false
 	}
-	if len(s.GeneratedStepbackInfo) > 0 {
-		return false
-	}
 	// If the other fields are set but not the ones above, the struct should be considered empty.
 	return true
 }
 
+// HasGeneratedStepbackInfo returns true if the StepbackInfo has generated stepback info.
```

Comment:
I think this comment can be more concise -- we can just keep the second sentence.